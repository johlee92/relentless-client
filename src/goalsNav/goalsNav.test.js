import React from 'react';
import ReactDOM from 'react-dom';
// because using Enzyme to test
// need to make sure the setupTest.js file links to adapter
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import GoalsNav from './goalsNav';
import { BrowserRouter } from 'react-router-dom';

describe(`GoalsNav component`, () => {
    const props = {
        className: 'test-class-name'
    }

    it('renders without crashing', () => {
        const div = document.createElement('div');
        ReactDOM.render(
            <BrowserRouter> 
                <GoalsNav />
            </BrowserRouter>, div);
        ReactDOM.unmountComponentAtNode(div);
    });

    it('renders the component by default', () => {
        const wrapper = shallow(<GoalsNav />)
        expect(toJson(wrapper)).toMatchSnapshot()
    })

    it('renders the component from props', () => {
        const wrapper = shallow(<GoalsNav {...props} />)
        expect(toJson(wrapper)).toMatchSnapshot()
    })
})
