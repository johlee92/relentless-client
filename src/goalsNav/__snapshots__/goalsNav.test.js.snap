// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GoalsNav component renders the component by default 1`] = `
<div
  className="GoalsNavSection"
>
  <div
    className="GoalsNav"
  >
    <button
      value="Invalid date"
    >
      Previous Week
    </button>
    <span>
      Week: 
    </span>
    <button
      value="Invalid date"
    >
      Next Week
    </button>
  </div>
  <div
    className="GoalsNav"
  >
    <button
      value="Annual"
    >
      Annual
    </button>
    <button
      value="Monthly"
    >
      Monthly
    </button>
    <button
      value="Weekly"
    >
      Weekly
    </button>
  </div>
</div>
`;

exports[`GoalsNav component renders the component from props 1`] = `
<div
  className="GoalsNavSection"
>
  <div
    className="GoalsNav"
  >
    <button
      value="Invalid date"
    >
      Previous Week
    </button>
    <span>
      Week: 
    </span>
    <button
      value="Invalid date"
    >
      Next Week
    </button>
  </div>
  <div
    className="GoalsNav"
  >
    <button
      value="Annual"
    >
      Annual
    </button>
    <button
      value="Monthly"
    >
      Monthly
    </button>
    <button
      value="Weekly"
    >
      Weekly
    </button>
  </div>
</div>
`;
